en/components/for_of.md

```diff
@@ -0,0 +1,97 @@
+The directive is creating scrollable containers and renders small chunks of the data. It is used inside `igx-grid` and it can be used to build virtual `igx-list`.
+`igxForOf` can be used to virtualize data horizontally, vertically or both.
+</p>
+<div class="divider"></div>
+
+### igxFor Demo
+
+<div class="sample-container loading" style="height:780px">
+    <iframe src='{environment:demosBaseUrl}/igx-for-sample-1' width="100%" height="100%" seamless frameBorder="0" onload="onSampleIframeContentLoaded(this);"></iframe>
+</div>
+<br/>
+<!--<button data-localize="stackblitz" class="stackblitz-btn">view on stackblitz</button> -->
+<div class="divider--half"></div>
+
+### Dependencies
+
+The `igxFor` directive is exported as as an `NgModule`, thus all you need to do in your application is to import the _IgxForOfModule_
+inside your `AppModule`:
+
+```typescript
+// app.module.ts
+
+import { IgxForOfModule } from 'igniteui-angular/main';
+
+@NgModule({
+    imports: [
+        ...
+        IgxForOfModule,
+        ...
+    ]
+})
+export class AppModule {}
+```
+
+### Usage
+
+Now that we have the `igxForOf` module imported, let’s get started with a basic configuration of the **igxFor** that binds to local data:
+
+```html
+<span #container>
+    <ng-template *igxFor="data"></ng-template>
+</span>
+```
+
+The **data** property is an array that provides the data.
+
+
+<div class="divider--half"></div>
+
+## API
+
+### Inputs
+
+Below is the list of all inputs that the developers may set to configure the `igxFor` look/behavior:
+| Name | Type | Description |
+| :--- |:--- | :--- |
+| id | string | Unique identifier of the Grid |
+| `igxForOf` | any[] | The data to be virtualized |
+| `igxForScrollOrientation` | string | Specifies the scroll direction - "horizontal" or "vertical", default value is _vertical_ |
+| `igxForScrollContainer` | any | Specifies the container where the helper scrollbars will be contained, it is useful when nesting the directive |
+| `igxForContainerSize` | any |  Specifies the container size in pixels. For "vertical" virtualization this is the height of the container. For "horizontal" virtualization this is the width of the container. |
+| `igxForItemSize` | any | Specifies the item size in pixels. For "vertical" virtualization this is the height of the item/row.  |
+
+<div class="divider--half"></div>
+
+### Outputs
+
+A list of the events emitted by the **igx-for**:
+
+| Name              | Description                                                              |
+| :---------------- | :----------------------------------------------------------------------- |
+| _Event emitters_  | _Notify for a change_                                                    |
+| `onChunkLoaded`   | Used when scrolled to emit the loaded chunk of data                      |
```

Comment:
@kdinev From what I know we're not using "ed"/"ing" events anymore. We have only one event to keep the event number lower across the product. Should we remove one of the events?

Reply:
@mpavlinov Correct. @rkaraivanov should chip in on what the correct naming should be. I guess `onChunkLoad`

Reply:
@kdinev @mpavlinov We're actually using both events at the moment in the grid. The "ing" event is used to clean some column/row states in certain scenarios, while the "ed" event is used for the keyboard navigation implementation.
From the directive's perspective itself we definitely need an event before the new chunk is actually loaded to provide  the users with the capability implement remote virtualization. If it's absolutely neccessary to move away from "ing"/"ed" naming convention then we could name one onScroll and the other onChunkLoad. Let me know what you think.

Reply:
`onChunkPreload` and `onChunkLoad`?